pkg/cluster/cluster_test.go

```diff
@@ -164,32 +162,45 @@ func TestClusterApplyDockerDesktopStartClusterOnly(t *testing.T) {
 
 	assert.Equal(t, false, f.d4m.started)
 	assert.Equal(t, 1, f.dockerClient.ncpu)
-	f.controller.Apply(context.Background(), &api.Cluster{
-		Product: string(ProductDockerDesktop),
-		MinCPUs: 0,
-	})
+	f, _ = controllerApply(f, ProductDockerDesktop, 0)
 	assert.Equal(t, true, f.d4m.started)
 	assert.Equal(t, 1, f.dockerClient.ncpu)
 }
 
+func controllerApply(f *fixture, product Product, cpus int) (*fixture, error) {
+	// We should to know that if the cpu <0 ,it's mean that we use the default value of CPU but may not equals to 0.
```

Comment:
I'm not sure the logic is good. I mean may I misunderstand something here. So, please check here.